discovery.go

```diff
@@ -29,7 +64,10 @@ func init() {
 
 func main() {
 	log.SetFlags(0)
-	handling.Setup()
+	viper.Set("etcd", mustHostOnlyURL(viper.GetString("etcd")))
+	viper.Set("host", mustHostOnlyURL(viper.GetString("host")))
+
+	handling.Setup(viper.GetString("etcd"))
```

Comment:
Usually I keep the flags/command line stuff read-only so it will only holds the original user input. Otherwise it's easy for code to get into situations where it forgets to sanitize the values and winds up working with the raw input. Up to you, though.